### PR TITLE
Feature/migration/291 phoenix idle

### PR DESCRIPTION
**Pull request by RaimundoCar**
- Add phoenix idle component:
-- 1) It uses one Idle to write last activity timestamp to local storage. If component is mounted and 2 hours has passed since last activity written in local storage, then it logouts current session
-- 2) It uses another Idle to time idle status. If 2 hours has passed with no activity, a dialog is shown. This dialog shows a timer of 30 seconds. If those 30 seconds run out, then it logouts current session

### CODE REVIEW

**Comment by RaimundoCar** [comment]
[(0.5) [Sprint 8]-[Administrar] Expiración de sesión en el cliente](https://trello.com/c/NXoTzn4L/291-05-sprint-8-administrar-expiraci%C3%B3n-de-sesi%C3%B3n-en-el-cliente)

File: javascript/components/phoenix-idle/index.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  // Function that shows logout warning dialog if LOGOUT_TIME has passed.
+  handleLogoutIdleChange ({ idle }){
+    if (!idle || this.state.showDialog || !Session.isLoggedIn() || Session.currentUserPrivileges() === 'root_admin') return;
+    this._clearLogoutInterval();
+    this.setState({showDialog: true, logoutInterval: setInterval(this.tick, 1000)})
+  }
+
+  hideIdleDialog (){
+    this._clearLogoutInterval();
+    this.setState({ showDialog: false }, () => { setTimeout(this.resetCoountdown, 2000) });
```
Comment (gegianno) [inline javascript/components/phoenix-idle/index.js]: `resetCoountdown`

File: javascript/components/phoenix-idle/index.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    var newCountdown = this.state.logoutCountdown - 1;
+    if(newCountdown < 0) {
+      this._clearLogoutInterval();
+      this.logOut();
+    } else {
+      this.setState({ logoutCountdown: newCountdown });
+    }
+  }
+
+  resetCoountdown (){
```
Comment (gegianno) [inline javascript/components/phoenix-idle/index.js]: `resetCoountdown `

File: javascript/components/phoenix-idle/index.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+      showDialog: false
+    };
+
+    this._clearLogoutInterval = this._clearLogoutInterval.bind(this);
+    this._clearTimeout = this._clearTimeout.bind(this);
+    this.logOut = this.logOut.bind(this);
+    this.handleLogoutIdleChange = this.handleLogoutIdleChange.bind(this);
+    this.handleStatusIdleChange = this.handleStatusIdleChange.bind(this);
+    this.hideIdleDialog = this.hideIdleDialog.bind(this);
+    this.resetCoountdown = this.resetCoountdown.bind(this);
```
Comment (gegianno) [inline javascript/components/phoenix-idle/index.js]: `resetCoountdown ` no falla porque está escrito así en todas partes jaja

Comment (RaimundoCar) [reply javascript/components/phoenix-idle/index.js]: Vaaale, on it

**Pull request merged**